deepchecks/recommender/checks/label_popularity_drift.py

```diff
@@ -68,28 +68,20 @@ def run_logic(self, context: Context) -> CheckResult:
             value: drift score.
             display: label distribution graph, comparing the train and test distributions.
         """
-        train_dataset = context.train.sample(self.n_samples, random_state=self.random_state)
-        test_dataset = context.test.sample(self.n_samples, random_state=self.random_state)
+        train_labels = context._y_pred_train
+        test_labels = context._y_pred_test
 
-        train_labels = pd.Series(train_dataset.label_col)
-        test_labels = pd.Series(test_dataset.label_col)
 
-        all_values = pd.Series(train_labels.tolist() + test_labels.tolist())
-        value_counts = all_values.value_counts(ascending=False)
-        value_counts.iloc[:] = list(range(len(value_counts)))
-        translation_dict = value_counts.to_dict()
+        # Compute item popularity for the training dataset
+        item_id = context._item_dataset.index_name
+        item_popularity = context._item_dataset.data.set_index(item_id)['train_popularity'].to_dict()
```

Comment:
How do we know that item_dataset has train_popularity as a filed in the dataset? It's a special field?

Reply:
fixed.
https://github.com/deepchecks/deepchecks/pull/2610/commits/fad11460e13a3dd3fbf60e45cceb214efebc8562